File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.08-pattern.js
function pattern_8() {
    let row = 5;
    let column = 5;

    for (let y = row; y > 0; y--) {
        for (let x = 1; x < column * 2; x++) {
            if (x <= column * 2 - column) {
                if (x < y) {
                    process.stdout.write(String(' '));
                } else {
                    process.stdout.write(String('*'));
                }
            } else if (x > column * 2 - column) {
                if (column * 2 - x < y) {
                    process.stdout.write(String(' '));
                } else {
                    process.stdout.write(String('*'));
                }
            }
        }

        console.log();
    }
}

pattern_8();

File: python/0.0-Basics/0.3-build-up-logical-thinking/0.3.08-pattern.py
from utils.clear import clear


def pattern_8():
    row = 5
    column = 10

    for i in range(row , 0, -1):
        for j in range(1, column):
            if j <= column /2 :
                if j >= i:
                    print("*", end="")
                else:
                    print(" ", end="")
            elif j > column / 2:
                key = column - j
                if key >= i:
                    print("*", end="")
                else:
                    print(" ", end="")
        print()

clear()
pattern_8()
